Framework/Geometry/src/Instrument.cpp

```diff
@@ -1315,18 +1248,17 @@ size_t Instrument::detectorIndex(const detid_t detID) const {
 /// DetectorInfo (masking, positions, rotations).
 boost::shared_ptr<ParameterMap> Instrument::makeLegacyParameterMap() const {
   auto pmap = boost::make_shared<ParameterMap>(*getParameterMap());
-  // Information will be stored directly in pmap so we do not need DetectorInfo
-  // or ComponentInfo.
-  pmap->setDetectorInfo(nullptr);
-  pmap->setComponentInfo(nullptr);
-  // Instrument is only needed for DetectorInfo access so it is not needed.
+  // Instrument is only needed for DetectorInfo access so it is not needed. This
+  // also clears DetectorInfo and ComponentInfo (information will be stored
+  // directly in pmap so we do not need them).
   pmap->setInstrument(nullptr);
 
-  if (!hasDetectorInfo() && !hasComponentInfo())
-    return pmap;
-
   const auto &baseInstr = m_map ? *m_instr : *this;
 
+  if (!getParameterMap()->hasDetectorInfo(&baseInstr) &&
```

Comment:
Do we need both checks? If the `ParameterMap` has one, it should have the other too. How about adding a `hasBeamline` rather than having both member functions?